### ISSUE TITLE
[BUG] Uncaught TypeError: Illegal invocation with react-dom

### ISSUE THREAD

**Issue by magicdawn** [violentmonkey/violentmonkey#1493]
- [x] I have searched for existing issues that already reported this problem and found none

**Describe the bug**
when react-dom is @required to a user script, 
react-dom will use `window.event`, and access `window.event` will cause `Illegal invocation`

https://github.com/facebook/react/blob/main/packages/react-dom/src/client/ReactDOMHostConfig.js#L371-L374

![image](https://user-images.githubusercontent.com/4067115/163679234-91b6499b-6810-4328-a3d3-bd38864a05e7.png)

and after I attach `window.__t__ = t`

![image](https://user-images.githubusercontent.com/4067115/163679275-520011c4-7f06-45c7-9e52-1b63d97583e9.png)

don't know how to trigger this behavior, after this error
following `setState` calls will fail

**To Reproduce**
Note that issues without a reproduction may be closed immediately!
1.
2.
3.

**Expected behavior**

**Actual behavior**

**What is the result in the [upcoming release](https://github.com/violentmonkey/violentmonkey/releases)?**
WARNING! Do an export in settings before testing!

**Environment:**
- OS: macOS
- Browser: chrome 
- Browser Version: 100.0.4896.127
- Violentmonkey Version:  2.13.0 & Violentmonkey BETA 2.13.0.18

**Additional context**

**Comment by magicdawn** [comment]
```
// @require       https://npm.elemecdn.com/react@18/umd/react.development.js
// @require       https://npm.elemecdn.com/react-dom@18/umd/react-dom.development.js
```

**Comment by tophf** [comment]
Works for me with this code:

```js
// ==UserScript==
// @name        test widow.event 
// @namespace   Violentmonkey Scripts
// @match       https://www.example.com/
// @require     https://npm.elemecdn.com/react@18/umd/react.development.js
// @require     https://npm.elemecdn.com/react-dom@18/umd/react-dom.development.js
// ==/UserScript==
window.onclick = () => console.log(window.event);
```

Please provide the exact code that reproduces the problem.

**Comment by magicdawn** [comment]
I have a small script 

// main.user.js
```js
// ==UserScript==
// @name          foo-bar-baz
// @namespace     https://magicdawn.fun
// @version       0.0.1
// @description   try to take over the world!
// @author        magicdawn
//
// @grant         GM_registerMenuCommand
// @grant         GM_notification
// @grant         unsafeWindow
// @grant         GM_xmlhttpRequest
// @grant         GM_getValue
// @grant         GM_setValue
// @grant         GM_deleteValue
//
// @include       https://www.google.com.hk/*
//
// @require       https://npm.elemecdn.com/axios@0.19.0/dist/axios.min.js
// @require       https://npm.elemecdn.com/jquery@3.4.1/dist/jquery.min.js
// @require       https://npm.elemecdn.com/react@18/umd/react.development.js
// @require       https://npm.elemecdn.com/react-dom@18/umd/react-dom.development.js
// ==/UserScript==

/******/ (() => { // webpackBootstrap
/******/ 	"use strict";
/******/ 	var __webpack_modules__ = ({

/***/ "./src/Test.tsx":
/*!**********************!*\
  !*** ./src/Test.tsx ***!
  \**********************/
/***/ ((__unused_webpack_module, __webpack_exports__, __webpack_require__) => {

__webpack_require__.r(__webpack_exports__);
/* harmony export */ __webpack_require__.d(__webpack_exports__, {
/* harmony export */   "default": () => (__WEBPACK_DEFAULT_EXPORT__)
/* harmony export */ });
/* harmony import */ var react__WEBPACK_IMPORTED_MODULE_0__ = __webpack_require__(/*! react */ "react");
/* harmony import */ var react__WEBPACK_IMPORTED_MODULE_0___default = /*#__PURE__*/__webpack_require__.n(react__WEBPACK_IMPORTED_MODULE_0__);


function Test() {
    const [c, setC] = (0,react__WEBPACK_IMPORTED_MODULE_0__.useState)(1);
    const onAdd = (0,react__WEBPACK_IMPORTED_MODULE_0__.useCallback)(() => {
        setC((c) => c + 1);
    }, []);
    const onMinus = (0,react__WEBPACK_IMPORTED_MODULE_0__.useCallback)(() => {
        setC((c) => c - 1);
    }, []);
    return (react__WEBPACK_IMPORTED_MODULE_0___default().createElement("div", null,
        react__WEBPACK_IMPORTED_MODULE_0___default().createElement("p", null,
            "current: ",
            c),
        react__WEBPACK_IMPORTED_MODULE_0___default().createElement("button", { style: { width: '30px', height: '20px' }, onClick: onMinus }, "-"),
        react__WEBPACK_IMPORTED_MODULE_0___default().createElement("button", { style: { width: '30px', height: '20px' }, onClick: onAdd }, "+")));
}
/* harmony default export */ const __WEBPACK_DEFAULT_EXPORT__ = (Test);


/***/ }),

/***/ "react":
/*!************************!*\
  !*** external "React" ***!
  \************************/
/***/ ((module) => {

module.exports = React;

/***/ }),

/***/ "react-dom":
/*!***************************!*\
  !*** external "ReactDOM" ***!
  \***************************/
/***/ ((module) => {

module.exports = ReactDOM;

/***/ })

/******/ 	});
/************************************************************************/
/******/ 	// The module cache
/******/ 	var __webpack_module_cache__ = {};
/******/ 	
/******/ 	// The require function
/******/ 	function __webpack_require__(moduleId) {
/******/ 		// Check if module is in cache
/******/ 		var cachedModule = __webpack_module_cache__[moduleId];
/******/ 		if (cachedModule !== undefined) {
/******/ 			return cachedModule.exports;
/******/ 		}
/******/ 		// Create a new module (and put it into the cache)
/******/ 		var module = __webpack_module_cache__[moduleId] = {
/******/ 			// no module.id needed
/******/ 			// no module.loaded needed
/******/ 			exports: {}
/******/ 		};
/******/ 	
/******/ 		// Execute the module function
/******/ 		__webpack_modules__[moduleId](module, module.exports, __webpack_require__);
/******/ 	
/******/ 		// Return the exports of the module
/******/ 		return module.exports;
/******/ 	}
/******/ 	
/************************************************************************/
/******/ 	/* webpack/runtime/compat get default export */
/******/ 	(() => {
/******/ 		// getDefaultExport function for compatibility with non-harmony modules
/******/ 		__webpack_require__.n = (module) => {
/******/ 			var getter = module && module.__esModule ?
/******/ 				() => (module['default']) :
/******/ 				() => (module);
/******/ 			__webpack_require__.d(getter, { a: getter });
/******/ 			return getter;
/******/ 		};
/******/ 	})();
/******/ 	
/******/ 	/* webpack/runtime/define property getters */
/******/ 	(() => {
/******/ 		// define getter functions for harmony exports
/******/ 		__webpack_require__.d = (exports, definition) => {
/******/ 			for(var key in definition) {
/******/ 				if(__webpack_require__.o(definition, key) && !__webpack_require__.o(exports, key)) {
/******/ 					Object.defineProperty(exports, key, { enumerable: true, get: definition[key] });
/******/ 				}
/******/ 			}
/******/ 		};
/******/ 	})();
/******/ 	
/******/ 	/* webpack/runtime/hasOwnProperty shorthand */
/******/ 	(() => {
/******/ 		__webpack_require__.o = (obj, prop) => (Object.prototype.hasOwnProperty.call(obj, prop))
/******/ 	})();
/******/ 	
/******/ 	/* webpack/runtime/make namespace object */
/******/ 	(() => {
/******/ 		// define __esModule on exports
/******/ 		__webpack_require__.r = (exports) => {
/******/ 			if(typeof Symbol !== 'undefined' && Symbol.toStringTag) {
/******/ 				Object.defineProperty(exports, Symbol.toStringTag, { value: 'Module' });
/******/ 			}
/******/ 			Object.defineProperty(exports, '__esModule', { value: true });
/******/ 		};
/******/ 	})();
/******/ 	
/************************************************************************/
var __webpack_exports__ = {};
// This entry need to be wrapped in an IIFE because it need to be isolated against other modules in the chunk.
(() => {
/*!**********************!*\
  !*** ./src/main.tsx ***!
  \**********************/
__webpack_require__.r(__webpack_exports__);
/* harmony export */ __webpack_require__.d(__webpack_exports__, {
/* harmony export */   "default": () => (/* binding */ main)
/* harmony export */ });
/* harmony import */ var react__WEBPACK_IMPORTED_MODULE_0__ = __webpack_require__(/*! react */ "react");
/* harmony import */ var react__WEBPACK_IMPORTED_MODULE_0___default = /*#__PURE__*/__webpack_require__.n(react__WEBPACK_IMPORTED_MODULE_0__);
/* harmony import */ var react_dom__WEBPACK_IMPORTED_MODULE_1__ = __webpack_require__(/*! react-dom */ "react-dom");
/* harmony import */ var react_dom__WEBPACK_IMPORTED_MODULE_1___default = /*#__PURE__*/__webpack_require__.n(react_dom__WEBPACK_IMPORTED_MODULE_1__);
/* harmony import */ var _Test__WEBPACK_IMPORTED_MODULE_2__ = __webpack_require__(/*! @Test */ "./src/Test.tsx");

// import { createRoot } from 'react-dom/client'


main();
async function main() {
    const logo = document.querySelector('img[alt="Google"]');
    if (!logo)
        return;
    const container = document.createElement('div');
    logo.parentElement?.appendChild(container);
    logo.remove();
    // const root = createRoot(container)
    // root.render(<Test></Test>)
    (0,react_dom__WEBPACK_IMPORTED_MODULE_1__.render)(react__WEBPACK_IMPORTED_MODULE_0___default().createElement(_Test__WEBPACK_IMPORTED_MODULE_2__["default"], null), container);
}

})();

/******/ })()
;
```

source code in a private repo, related files here

```tsx
// main.tsx
import React from 'react'
// import { createRoot } from 'react-dom/client'
import { render } from 'react-dom'

import Test from '@Test'

main()
export default async function main() {
  const logo = document.querySelector('img[alt="Google"]')
  if (!logo) return

  const container = document.createElement('div')
  logo.parentElement?.appendChild(container)

  logo.remove()

  // const root = createRoot(container)
  // root.render(<Test></Test>)

  render(<Test></Test>, container)
}
```


```tsx
// Test.tsx
import React from 'react'
import { useCallback, useState } from 'react'

function Test() {
  const [c, setC] = useState(1)

  const onAdd = useCallback(() => {
    setC((c) => c + 1)
  }, [])
  const onMinus = useCallback(() => {
    setC((c) => c - 1)
  }, [])

  return (
    <div>
      <p>current: {c}</p>
      <button style={{ width: '30px', height: '20px' }} onClick={onMinus}>
        -
      </button>
      <button style={{ width: '30px', height: '20px' }} onClick={onAdd}>
        +
      </button>
    </div>
  )
}

export default Test
```


this main.user.js will remove https://www.google.com.hk/ `google logo`
and a counter with `+` / `-`,

to reproduce
- navigate to  https://www.google.com.hk/
- click `+` , see 1 -> 2
- click `+` again, this error happens

![image](https://user-images.githubusercontent.com/4067115/163681987-8ceb25da-974f-4b35-a736-db9175e039dd.png)

**Comment by magicdawn** [comment]
it's not the `window.event` error, but something else related to react. 
don't know.

**Comment by magicdawn** [comment]
same main.user.js work in TamperMonkey

![image](https://user-images.githubusercontent.com/4067115/163682397-b52cb922-8d90-4a52-955d-b28585972f9f.png)

**Comment by magicdawn** [comment]
source repo
https://github.com/magicdawn/violentmonkey-react-issue

**Comment by tophf** [comment]
Thanks, fixed in bc08540a.
Test build: [zip](https://github.com/violentmonkey/violentmonkey/suites/6146027042/artifacts/214595761)
